R/zzz.R

```diff
@@ -9,10 +17,43 @@
     packageStartupMessage("- CmdStan path set to: ", cmdstan_path(), "")
     packageStartupMessage("- Use set_cmdstan_path() to change the path")
   }
-}
 
-.onLoad <- function(...) {
-  cmdstanr_initialize()
+  latest_version <- try(suppressWarnings(latest_released_version()), silent = TRUE)
+  if (!inherits(latest_version, "try-error")
+      && latest_version > cmdstan_version()) {
+      packageStartupMessage(
+        "\nA newer version of CmdStan is available.",
+        " See ?install_cmdstan() to install it."
+      )
+  }
 }
 
 
+cmdstanr_initialize <- function() {
```

Comment:
This function is all the same, I just moved it to zzz.R from path.R since it's only used in `.onLoad()`